src/riak_kv_bucket.erl

```diff
@@ -43,6 +44,43 @@
 
 -export_type([props/0]).
 
+-define(VALID_PROPERTIES,
+        ["allow_mult",
+         "basic_quorum",
+         "big_vclock",
+         "bucket_type",
+         "chash_keyfun",
+         "dvv_enabled",
+         "dw",
+         "last_write_wins",
+         "linkfun",
+         "n_val",
+         "notfound_ok",
+         "old_vclock",
+         "postcommit",
+         "pr",
+         "precommit",
+         "pw",
+         "r",
+         "rw",
+         "small_vclock",
+         "w",
+         "write_once",
+         "young_vclock",
+         %% TS-specific ones:
+         "ddl",
+         "table_def"
+        ]).
+
+-spec is_valid_property(string() | binary()) -> boolean().
```

Comment:
Quite simply, there's a list_to_existing_atom down the line, and here's why
I attempt to collect all possible values for the bucket properties here.
On 16 Mar 2016 8:24 pm, "John Daily" <notifications@github.com> wrote:

> In src/riak_kv_bucket.erl
> <https://github.com/basho/riak_kv/pull/1369#discussion_r56388669>:
>
> > +         "postcommit",
> > +         "pr",
> > +         "precommit",
> > +         "pw",
> > +         "r",
> > +         "rw",
> > +         "small_vclock",
> > +         "w",
> > +         "write_once",
> > +         "young_vclock",
> > +         %% TS-specific ones:
> > +         "ddl",
> > +         "table_def"
> > +        ]).
> > +
> > +-spec is_valid_property(string() | binary()) -> boolean().
>
> Tough call. We've not constrained bucket properties in the past, and it's
> unclear this is a good reason to start doing so.
>
> On the other hand, we've certainly wanted to document all possible bucket
> properties to get out of the JSON-on-the-CLI business for riak-admin
> bucket-type create, so this work may prove valuable even if we choose not
> to use it.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/basho/riak_kv/pull/1369/files/704efa9a70dc8148468ddf16bc74504aa6494c22#r56388669>
>